### PR TITLE
Add classroom session list features

### PR DESCRIPTION
**Pull request by cacadosman**

### CODE REVIEW

File: app/src/main/java/blibli/future/pratica/adapter/ClassroomSessionAdapter.kt [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+        val currentTimestamp = Timestamp(Date().time)
+        sessions[position]?.let {session ->
+            val hasChecked: Boolean = currentTimestamp.after(Timestamp(session.startTime))
+            if (hasChecked) {
+                holder.checkImage.setImageResource(R.drawable.ic_check_circle_black_24dp)
+            }
+
+            val date = Timestamp(session.startTime)
+            var content = SimpleDateFormat("yyyy/MM/dd HH:mm").format(date)
```
Comment (bayoharyo) [inline app/src/main/java/blibli/future/pratica/adapter/ClassroomSessionAdapter.kt]: put date format to Constant

File: app/src/main/java/blibli/future/pratica/adapter/ModuleRequestAdapter.kt [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+    override fun getItemCount(): Int = moduleRequest.size
+
+    override fun onBindItemViewHolder(holder: RecyclerView.ViewHolder, position: Int) {
+        holder as ModuleRequestViewHolder
+        val item = moduleRequest.get(position)
+        item?.let {
+            holder.title.text = item.title
+            holder.author.text = "Submitted by : " + item.user
+
+            val sdf = java.text.SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss'Z'")
```
Comment (bayoharyo) [inline app/src/main/java/blibli/future/pratica/adapter/ModuleRequestAdapter.kt]: change val name

File: app/src/main/java/blibli/future/pratica/view/fragment/ModuleRequestFragment.kt [hunk 3]
```diff
@@ -1,33 +1,84 @@
...
+        val likedArgs = Bundle()
+        likedArgs.putBoolean("liked", true)
+
+        val allModuleRequestFragment = getFragment(RvModuleRequestFragment::class.java.name)
+        val popularModuleRequestFragment = getFragment(RvModuleRequestFragment::class.java.name)
+        val likedModuleRequestFragment = getFragment(RvModuleRequestFragment::class.java.name)
+        popularModuleRequestFragment.arguments = popularArgs
+        likedModuleRequestFragment.arguments = likedArgs
+
+        val fragmentList = mutableListOf<TabFragment>()
```
Comment (bayoharyo) [inline app/src/main/java/blibli/future/pratica/view/fragment/ModuleRequestFragment.kt]: use HashMap

File: app/src/main/java/blibli/future/pratica/view/activity/ClassroomDetailActivity.kt [hunk 4]
```diff
@@ -4,22 +4,35 @@ import android.content.Context
...
 import javax.inject.Inject
 
 class ClassroomDetailActivity : BaseActivity() {
 
     @Inject lateinit var factory: ViewModelProvider.Factory
     val viewModel by viewModels<TraineeClassroomDetailViewModel>{factory}
     lateinit var binding: ActivityClassroomDetailBinding
+    lateinit var sessionAdapter: ClassroomSessionAdapter
+    lateinit var materialAdapter: ClassroomMaterialAdapter
+    lateinit var sessionRecyclerView: RecyclerView
```
Comment (bayoharyo) [inline app/src/main/java/blibli/future/pratica/view/activity/ClassroomDetailActivity.kt]: remove sessionRecyclerView and materialRecyclerView from global variable, can be accessed from binding.

File: app/src/main/java/blibli/future/pratica/view/activity/ClassroomDetailActivity.kt [hunk 5]
```diff
@@ -31,6 +44,26 @@ class ClassroomDetailActivity : BaseActivity() {
...
+        sessionRecyclerView = binding.sessionRecyclerView
+        sessionRecyclerView.layoutManager = sessionLayoutManager
+
+        val materialLayoutManager = LinearLayoutManager(this)
+        materialRecyclerView = binding.materialRecyclerView
+        materialRecyclerView.layoutManager = materialLayoutManager
+
+        viewModel.classroomDetail.observe(this, Observer {
+            it.sessions?.let {
+                sessionAdapter = ClassroomSessionAdapter(it.toMutableList())
```
Comment (bayoharyo) [inline app/src/main/java/blibli/future/pratica/view/activity/ClassroomDetailActivity.kt]: move adapter initialization outside view model observe.

File: app/src/main/java/blibli/future/pratica/view/activity/ClassroomDetailActivity.kt [hunk 6]
```diff
@@ -31,6 +44,26 @@ class ClassroomDetailActivity : BaseActivity() {
...
+                sessionRecyclerView.adapter = sessionAdapter
+            }
+
+            it.materials?.let {
+                materialAdapter = ClassroomMaterialAdapter(it.toMutableList())
```
Comment (bayoharyo) [inline app/src/main/java/blibli/future/pratica/view/activity/ClassroomDetailActivity.kt]: move adapter initialization outside view model observe

File: app/src/main/java/blibli/future/pratica/adapter/ClassroomMaterialAdapter.kt [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+import blibli.future.pratica.BuildConfig
+import blibli.future.pratica.R
+import blibli.future.pratica.adapter.base.BaseAdapter
+import blibli.future.pratica.network.response.Material
+import kotlinx.android.synthetic.main.classroom_material_item.view.*
+
+class ClassroomMaterialAdapter (var materials: MutableList<Material?> = mutableListOf()) :
+    BaseAdapter<Material>(materials){
+
+    override fun onBindItemViewHolder(holder: RecyclerView.ViewHolder, position: Int) {
```
Comment (bayoharyo) [inline app/src/main/java/blibli/future/pratica/adapter/ClassroomMaterialAdapter.kt]: change type of holder to MaterialViewHolder so you don't need to type cast

File: app/src/main/java/blibli/future/pratica/adapter/ClassroomResultDoneAdapter.kt [hunk 8]
```diff
@@ -0,0 +1,58 @@
...
+
+    override fun onBindItemViewHolder(holder: RecyclerView.ViewHolder, position: Int) {
+        holder as ClassroomResultViewHolder
+        val item = classrooms.get(position)
+        item?.let {
+            holder.name.text = it.name
+            holder.trainer.text = "Trainer : " + it.trainer
+
+            if ( item.hasExam ?: false ) {
+                holder.score.text = it.score.toString()
```
Comment (bayoharyo) [inline app/src/main/java/blibli/future/pratica/adapter/ClassroomResultDoneAdapter.kt]: extract to separate function

File: app/src/main/java/blibli/future/pratica/adapter/ClassroomResultDoneAdapter.kt [hunk 9]
```diff
@@ -0,0 +1,58 @@
...
+                holder.minScore.text = "Min " + it.minScore.toString()
+                holder.passed.text = it.status.toString()
+            } else {
+                holder.score.text = "--"
```
Comment (bayoharyo) [inline app/src/main/java/blibli/future/pratica/adapter/ClassroomResultDoneAdapter.kt]: extract to separate function

File: app/src/main/java/blibli/future/pratica/view/fragment/ClassroomResultFragment.kt [hunk 10]
```diff
@@ -0,0 +1,81 @@
...
+        setHasOptionsMenu(true)
+        binding = DataBindingUtil.inflate(
+            inflater, R.layout.classroom_result_fragment, container, false)
+        return binding.root
+    }
+
+    override fun onActivityCreated(savedInstanceState: Bundle?) {
+        super.onActivityCreated(savedInstanceState)
+
+        (activity as AppCompatActivity).supportActionBar?.title = "Followed Classroom"
```
Comment (bayoharyo) [inline app/src/main/java/blibli/future/pratica/view/fragment/ClassroomResultFragment.kt]: (activity as AppCompatActivity).supportActionBar?.apply {

   title = "Followed Classroom"
   setDisplayHomeAsUpEnabled(true)
}

File: app/src/main/java/blibli/future/pratica/view/fragment/ClassroomResultFragment.kt [hunk 11]
```diff
@@ -0,0 +1,81 @@
...
+        (activity as AppCompatActivity).supportActionBar?.setDisplayHomeAsUpEnabled(true)
+
+        val ongoingArgs = Bundle()
+        ongoingArgs.putString("status", "ongoing")
+
+        val doneArgs = Bundle()
+        doneArgs.putString("status", "done")
+
+        val ongoingClassroomFollowedFragment = getFragment(ClassroomResultListFragment::class.java.name)
```
Comment (bayoharyo) [inline app/src/main/java/blibli/future/pratica/view/fragment/ClassroomResultFragment.kt]: `val ongoingClassroomFollowedFragment = getFragment(ClassroomResultListFragment::class.java.name).apply {
      arguments = Bundle().putString("status", "ongoing")
}`

Comment (bayoharyo) [reply app/src/main/java/blibli/future/pratica/view/fragment/ClassroomResultFragment.kt]: Please apply this also for `doneClassroomFollowedFragment`

Comment (bayoharyo) [reply app/src/main/java/blibli/future/pratica/view/fragment/ClassroomResultFragment.kt]: create constant for Classroom status example access : 
ClassroomStatus.ONGOING
ClassroomStatus.DONE

File: app/src/main/java/blibli/future/pratica/adapter/ClassroomSessionAdapter.kt (hunk 1 above)
Comment (cacadosman) [reply app/src/main/java/blibli/future/pratica/adapter/ClassroomSessionAdapter.kt]: DONE

File: app/src/main/java/blibli/future/pratica/adapter/ModuleRequestAdapter.kt (hunk 2 above)
Comment (cacadosman) [reply app/src/main/java/blibli/future/pratica/adapter/ModuleRequestAdapter.kt]: DONE

File: app/src/main/java/blibli/future/pratica/view/activity/ClassroomDetailActivity.kt (hunk 5 above)
Comment (cacadosman) [reply app/src/main/java/blibli/future/pratica/view/activity/ClassroomDetailActivity.kt]: DONE

File: app/src/main/java/blibli/future/pratica/view/activity/ClassroomDetailActivity.kt (hunk 6 above)
Comment (cacadosman) [reply app/src/main/java/blibli/future/pratica/view/activity/ClassroomDetailActivity.kt]: DONE

File: app/src/main/java/blibli/future/pratica/view/fragment/ClassroomResultFragment.kt (hunk 10 above)
Comment (cacadosman) [reply app/src/main/java/blibli/future/pratica/view/fragment/ClassroomResultFragment.kt]: DONE

File: app/src/main/java/blibli/future/pratica/view/fragment/ClassroomResultFragment.kt (hunk 11 above)
Comment (cacadosman) [reply app/src/main/java/blibli/future/pratica/view/fragment/ClassroomResultFragment.kt]: DONE

**Pull request merged**